### PR TITLE
Disable buffer swapping

### LINKED ISSUE #5734: Very slow rendering when using `autoswap=True` in the VisPy Canvas
## 🐛 Bug

I was really puzzled why the brush was not so responsive as it should have even on low-resolution images. I measured FPS and discovered that it never exceeded 10-15 when you use the brush even on very small images (e.g. `examples\add_labels.py`).

Initially, I thought that it was because of some very inefficient code in the Label layer, but after long hours with a debugger and profiler I eliminated this hypothesis. There is certainly room for improvement, but there is no such code that can limit FPS to 15 even on 100x100 images.

Finally, I found the culprit of this issue, it was because of using `autoswap=True` in the VisPy Canvas. Setting `autoswap` to `False` in the following code immediately increases the FPS from 13 to 50 in `examples\add_labels.py` on my laptop:
https://github.com/napari/napari/blob/cf55aa005230a89b906be2447ce900be2c1f200a/napari/_qt/qt_viewer.py#L431-L436

Calling `Canvas.swap_buffers` at each update below blocks the OpenGL engine at least for 50-60 ms on my laptop not depending on the resolution of an image causing a significant drop in the rendering performance:
https://github.com/vispy/vispy/blob/5449e3402b35da4d6a8215d8d72e19dcf05917d7/vispy/app/canvas.py#L236-L237
which leads to the execution of the following code:
https://github.com/vispy/vispy/blob/5449e3402b35da4d6a8215d8d72e19dcf05917d7/vispy/app/backends/_qt.py#L869

I didn't dig any deeper into VisPy as disabling `autoswap` solves the issue without any noticeable bugs so far for me.

## Environment

I tested it on a laptop with Windows and integrated graphics:

```
napari: 0.5.0a2.dev55+g8a793b9f.d20230406
Platform: Windows-10-10.0.22621-SP0
Python: 3.9.1 (default, Dec 11 2020, 09:29:25) [MSC v.1916 64 bit (AMD64)]
Qt: 5.12.9
PyQt5: 5.12.3
NumPy: 1.22.4
SciPy: 1.9.1
Dask: 2023.3.2
VisPy: 0.12.2
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.7
app-model: 0.1.2
npe2: 0.6.2

OpenGL:
- GL version: 4.6.13596 Compatibility Profile Context 20.10.22.14 27.20.11022.14001
- MAX_TEXTURE_SIZE: 16384
```

### PR DESCRIPTION
**Pull request by brisvag**
# Fixes/Closes
Closes #5734.

# Description
This change drastically improves performance on some hardware by sacrificing some quality (tearing is more likely). However, several people tested it and didn't notice any appreciable degradation of the rendering, so it seems like a worthwile exchange. See #5734 for more details.

We might be able to fix this on the vispy side, but for now this works well on the napari side.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

cc @ksofiyuk

### CODE REVIEW

**Comment by Czaki** [comment]
Could we control this using environment variable?

**Comment by brisvag** [comment]
> Could we control this using environment variable?

I would make it disabled by default in that case, though. Can you point me to where the env variable stuff is normally set?

**Comment by Czaki** [comment]
The user could set it in his .bashrc, .zshrc .profile, etc

**Comment by brisvag** [comment]
I meant where in the napari code we read all the env variables and store them for later usage in the code :P Or would you just read it when the canvas is created?

**Comment by Czaki** [comment]
I do not know the such place. 

I prefer using it when creating Canvas. It is rare enough for me not to think about performance.

**Comment by brisvag** [comment]
Something like this? I suppose this should be somewhere in the docs as well.

File: napari/_qt/qt_viewer.py [hunk 1]
```diff
@@ -433,6 +434,7 @@ def _create_canvas(self) -> None:
             vsync=True,
             parent=self,
             size=self.viewer._canvas_size[::-1],
+            autoswap=os.environ.get("NAPARI_AUTOSWAP"),  # see #5734
```
Comment (Czaki) [inline napari/_qt/qt_viewer.py]:
```suggestion
            autoswap=os.environ.get("NAPARI_AUTOSWAP", "0") == "1",  # see #5734
```
Better is check for given water than presence.

**Comment by Czaki** [comment]
Yes. This needs to be added to docs.

**Comment by brisvag** [comment]
I see some mentions of environment variable regarding rendering in `docs/guides/rendering.md`, as well as `docs/guides/rendering-explanation.md`. I suppose one of those places could work? Opinions from the docs experts? @melissawm @psobolewskiPhD

**Comment by Czaki** [comment]
I prefer to have a single place with list of all Environment variables use by napari. Best, next to settings.

**Comment by psobolewskiPhD** [comment]
I think this section is prolly the best? But I think the whole thing at some point needs to be rewritten—I think it's mostly autogenerated?
https://napari.org/stable/guides/preferences.html
I mean you could also add this experimental preferences pane, like the async stuff? 🤷‍♂️

**Comment by brisvag** [comment]
I moved it to the experimental settings. Now it works both from there and as a env variable!

**Comment by brisvag** [comment]
> I think it's mostly autogenerated?

Yep, so this should get added automatically now that it's a preference.

Maybe experimental is not quite the right place (advanced? :P) but for now I think it works.

**Comment by ksofiyuk** [comment]
I've tested it, works great for me, @brisvag thank you for the fix!

**Review by psobolewskiPhD — APPROVED** [review]
I can't see any performance difference nor any visual effect, but I like the implementation.
I wonder whether it should be true or false by default, given that the existing behavior is true.

**Comment by brisvag** [comment]
My reasoning was: it's more likely that a new user affected by the bug will simply think "oh well napari has terrible performance, bye" rather than opening an issue or testing experimental settings. And the cost seems basically zero otherwise :shrug:

**Review by alisterburt — APPROVED** [review]
+1 on `False` as the new default - if we get reports of tearing we can reevaluate

**Pull request merged**